src/hotspot/share/jvmci/jvmci.hpp

```diff
@@ -163,6 +163,115 @@ class JVMCI : public AllStatic {
   static void event2(const char* format, ...) ATTRIBUTE_PRINTF(1, 2);
   static void event3(const char* format, ...) ATTRIBUTE_PRINTF(1, 2);
   static void event4(const char* format, ...) ATTRIBUTE_PRINTF(1, 2);
+
+  static jlong* _jvmci_old_thread_counters;
+  static void collect_counters(jlong* array, int length);
+
+  static bool resize_counters(JavaThread* thread, int current_size, int new_size);
+
+  static bool resize_all_jvmci_counters(int new_size);
+
+  static void free_counters(JavaThread* thread);
+
+  static void accumulate_counters(JavaThread* thread);
+
+  static void init_counters();
+
+  static void free_counters();
+};
+
+
```

Comment:
There seems to be a crazy amount of whitespace here.

Reply:
I'd meant to go back and pretty this up.  I was just dumping things in there as I migrated them from elsewhere.  I've grouped them with some short comments and tried to clean it up.  I also fixed some visibility issues and removed the use of friend for JavaThread.